### PR TITLE
feat(react-native-test-app-msal): MSAL module for react-native-test-app

### PR DESCRIPTION
**Pull request by tido64**
### Description

Extracted MSAL support module from our internal repo.

### Remaining work

- [ ] Android: JetPack Compose?
- [x] iOS
  - [x] ~~Clear out cookies from the WebView used for auth~~ We can wipe out the account, but not the cookies it looks like. Something to investigate for later.
  - [x] https://github.com/microsoft/react-native-test-app/pull/560
- [x] macOS: Since the switcher is written in SwiftUI, it just needs some adjustments/testing
- [ ] Windows: This will be a new implementation since we don't have one from before
- [ ] Documentation

### Test plan

| Account switcher | List of accounts from MSAL | Adding a new account |
|-|-|-|
| Simulator Screen Shot - iPhone 12 Pro - 2021-10-15 at 17 06 04 | Simulator Screen Shot - iPhone 12 Pro - 2021-10-15 at 17 07 09 | Simulator Screen Shot - iPhone 12 Pro - 2021-10-15 at 17 08 29 |

### CODE REVIEW

**Comment by afoxman** [comment]
If this is a module for RNTA, should it be developed and published from that repo instead of rnx-kit?

File: packages/react-native-test-app-msal/ios/ReactTestApp-MSAL.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -0,0 +1 @@
+/* Dummy file so @react-native-community/cli recognizes this as an iOS package */
```
Comment (kelset) [inline packages/react-native-test-app-msal/ios/ReactTestApp-MSAL.xcodeproj/project.pbxproj:1]: 🤣

**Review by kelset — APPROVED** [review]
I think it's ok to have this package live in rnx-kit since it fits the "range" of plugins and tools we create for RN experiences. It just so happens that this one is specifically for RNTA

File: packages/react-native-test-app-msal/ios/ReactTestApp-MSAL.xcodeproj/project.pbxproj (hunk 1 above)
Comment (tido64) [reply packages/react-native-test-app-msal/ios/ReactTestApp-MSAL.xcodeproj/project.pbxproj:1]: Yeah, it's on my wishlist to the cli team that they should look for `Podfile`.

**Pull request merged**